piptools/writer.py

```diff
@@ -230,6 +232,8 @@ def _format_requirement(
         ireq_hashes = (hashes if hashes is not None else {}).get(ireq)
 
         line = format_requirement(ireq, marker=marker, hashes=ireq_hashes)
+        if self.constraints_compatible:
+            line = re.sub(r"\[.*\]", "", line)
```

Comment:
I'm curious if this is enough: what if there will be duplicate entries because of the same name with different extras being present?

Reply:
what do you mean? any example?

Reply:
This will not happen because it would translate as a duplicate requirement, not allowed by pip. Multiple extras are listed on the same line `name[one,two]` and not separated.